test/built-ins/RegExp/named-groups/functional-replace.js

```diff
@@ -0,0 +1,154 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+  Function argument to String.prototype.replace gets groups as the
+  third-to-last argument
+esid: pending
+features: [regexp-named-groups]
+---*/
+
+// --- Unicode mode ---
+
+// @@replace with a callable replacement argument (no named captures).
+{
+  let result = "abcd".replace(/(.)(.)/u, (match, fst, snd, offset, str) => {
```

Comment:
the `{}` wrappers are a perfect indication these tests should be split into multiple files. There's already the description for each part.

It would be great to provide a `info` tag containing the spec parts allowing each cases. It's a valuable reference to match the validity for each assertion.